### DISCUSSION TITLE
Accurancy

### DISCUSSION THREAD

**Discussion by vittoria-thomasini** [ChenglongMa/SkinToneClassifier#9] [Q&A]
Hi ChenglongMa,

How you calculate the accuracy ? Are you using some study as reference for this ?

**Comment by ChenglongMa** [comment]
Hi @vittoria-thomasini,

Thanks for your question.

This is our original evaluation metric.

At the very beginning, we used a **weighted** Delta E (CIE 2000) ($\Delta E_{00}$) distance metric ([see here](https://en.wikipedia.org/wiki/Color_difference#CIEDE2000)) to measure the accuracy:

$$c^*=\underset{c\in C}{\arg \min} \sum_{d \in D}{\rho_d \cdot \Delta E_{00}(d,c)},$$
where
* $C$ is the set of color **C**ategories in the specified palette. $c \in C$.
* $D$ is the set of detected **D**ominant colors in the specified portrait/image. $d \in D$.
* $\rho_d$ is the percentage of the skin area that the color $d$ occupies.

**It measures the distance between the assigned color palette and the dominant colors in the face area. Values are between 0 and 100. A lower value indicates a more accurate classification.**

However, sometimes the word **Distance** is not easy for non-technical users to understand, so I changed it to **Accuracy**.
Accordingly, the formula becomes:

$$accuracy = 100 - c^*$$.

Then, the values are still between 0 and 100, but the bigger the better.

Thanks!

We also explained this metric in [our paper](https://doi.org/10.1111/ssqu.13242).
